src/Features/ExternalAccess/Copilot/Completion/IContextItem.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Copilot.Completion;
+
+[JsonDerivedType(typeof(CodeSnippetItem))]
+[JsonDerivedType(typeof(TraitItem))]
```

Comment:
ni: call these CodeSnippetContextItem and TraitContextItem to make it clearer what they are.